soda/cmd/validate/model.go

```diff
@@ -0,0 +1,111 @@
+package validate
+
+import (
+	"fmt"
+	"go/ast"
+	"regexp"
+	"strings"
+)
+
+type model struct {
+	packages map[string]*ast.Package
+	tags     map[string][]string
+	rules    []Checker
+}
+
+type Checker interface {
+	Check(tag string, structName string) []ValidationError
+}
+
+type ValidationError struct {
+	invalidSymbols string
+	field          string
+	structName     string
+	duplicate      bool
+}
+
+func (e *ValidationError) Error() string {
+	return fmt.Sprintf("Invalid symbols '%v' contained in %v.%v", e.invalidSymbols, e.structName, e.field)
+}
+
+type Rule struct {
```

Comment:
Another type who's fields should be exported so it can be used programmatically.